sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/StorageBearerTokenChallengeAuthorizationPolicy.java

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.policy;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.util.CoreUtils;
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The storage authorization policy which supports challenge.
+ */
+public class StorageBearerTokenChallengeAuthorizationPolicy extends BearerTokenAuthenticationPolicy {
+
+    private static final String DEFAULT_SCOPE = "/.default";
+    private String[] scopes;
+
+    private static final Pattern AUTHENTICATION_CHALLENGE_PATTERN
```

Comment:
Does a complicated pattern need to be used here? It seems that the Storage bearer token format is well-known and well defined:

https://learn.microsoft.com/rest/api/storageservices/authorize-with-azure-active-directory#sample-response-to-bearer-challenge

Could this be simplified to make understanding what is happening easier?

Reply:
Wasn't sure what the best way to parse it was so I just based it off of the ArmChallenge policy. I'd def be open to simplifying it, do you have any suggestions on what should be done?